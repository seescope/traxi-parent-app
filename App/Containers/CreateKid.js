import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Image, TouchableOpacity, Alert } from 'react-native';
import ImagePicker from 'react-native-image-picker';
import { Actions } from 'react-native-router-flux';

import { enterKidName, selectKidImage } from '../Actions/Actions';
import setupKid from '../Actions/SetupKid';
import watchDevice from '../Actions/WatchDevice';
import Background from '../Components/Background';
import HeaderText from '../Components/HeaderText';
import Spacing from '../Components/Spacing';
import BodyText from '../Components/BodyText';
import SmallText from '../Components/SmallText';
import TextInput from '../Components/TextInput';
import { TRAXI_BLUE } from '../Constants/Colours';
import { isIOS, logError } from '../Utils';

const getSource = response => {
  let source;

  if (response.data) {
    if (isIOS) {
      source = { uri: response.uri.replace('file://', ''), isStatic: true };
    } else {
      source = { uri: response.uri, isStatic: true };
    }
  }

  return source;
};

export const selectImage = dispatch => {
  const options = {
    title: 'Select Image',
    storageOptions: {
      skipBackup: true,
    },
  };

  ImagePicker.launchImageLibrary(options, response => {
    if (response.didCancel) {
      return;
    }

    if (response.error) {
      if (response.error === 'Photo library permissions not granted') {
        Alert.alert(
          'Unable to access your photos',
          'Please allow traxi to access your photos to continue.'
        );
      }

      logError(response.error);
      return;
    }

    const source = getSource(response);
    dispatch(selectKidImage(source.uri));

    Actions.walkthrough();

    dispatch(setupKid()).then(() => {
      dispatch(watchDevice());
    });
  });
};

const style = {
  container: {
    backgroundColor: TRAXI_BLUE,
    flex: 1,
    padding: 16,
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
};

const CreateKid = ({ parentName, onChangeText, onPress }) => (
  <Background style={style.container}>
    <HeaderText>Add your child, {parentName}</HeaderText>

    <Spacing />

    <BodyText>
      Type your child's name in the box below, then tap on the circle
      to add a picture.
    </BodyText>

    <Spacing />

    <SmallText>
      Don't worry, only you can see your child's picture. If you feel
      uncomfortable using their photo, you can use any image you like.
    </SmallText>

    <Spacing height={36} />

    <TextInput placeholder="eg. Jen Smith" onChangeText={onChangeText} />

    <Spacing height={16} />

    <TouchableOpacity onPress={onPress} style={style.deviceContainer}>
      <Image source={require('../Images/add_image.png')} />
    </TouchableOpacity>
  </Background>
);

CreateKid.propTypes = {
  parentName: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired,
  onPress: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  parentName: state.parentName,
});

const mapDispatchToProps = dispatch => ({
  onChangeText: text => dispatch(enterKidName(text)),
  onPress: () => selectImage(dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(CreateKid);
