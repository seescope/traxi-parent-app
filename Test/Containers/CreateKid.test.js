import { View } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { connect, Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import CreateKid from '../../App/Containers/CreateKid';

// Why? Why!?
global.Promise = require.requireActual('promise');

jest.mock('../../App/Actions/LoginWithMethod', () => {
  return jest.fn(() => {
    return (dispatch) => {
      return new Promise.resolve({ type: 'SOMETHIGN' });
    }
  });
});

jest.mock('react-native-router-flux', () => {
  mockFindKid = jest.fn();
  return {
    Actions: {
      findKid: mockFindKid
    }
  };
});

jest.mock('../../App/Components/Background');

const mockStore = configureStore([thunk]);
const testStore = mockStore({ parentName: 'Name'});

const CreateKidComponent = () => (
  <Provider store={testStore}>
    <CreateKid />
  </Provider>
);

it('renders the <CreateKid> component', () => {
  const tree = renderer.create(
    <CreateKidComponent />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
