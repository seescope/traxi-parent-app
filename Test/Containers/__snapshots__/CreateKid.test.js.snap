exports[`test handles when the image is selected 1`] = `
Array [
  Array [
    Object {
      "type": "RESET_STATE",
    },
  ],
  Array [
    Object {
      "avatarURL": "heythere",
      "type": "SELECT_KID_IMAGE",
    },
  ],
  Array [
    [Function],
  ],
  Array [
    [Function],
  ],
]
`;

exports[`test renders the <CreateKid> component 1`] = `
<View
  colors={
    Array [
      "#448AFF",
      "#90CAF9",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#448AFF",
      "flex": 1,
      "justifyContent": "flex-start",
      "padding": 16,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-ExtraBold",
        "fontSize": 30,
        "marginBottom": -2,
        "textAlign": "center",
      }
    }>
    Add your child, 
    Name
  </Text>
  <View
    style={
      Object {
        "height": 24,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-Regular",
        "fontSize": 18,
        "fontWeight": "200",
        "textAlign": "center",
      }
    }>
    Type your child\'s name in the box below, then tap on the circle to add a picture.
  </Text>
  <View
    style={
      Object {
        "height": 24,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontSize": 14,
        "fontWeight": "400",
        "textAlign": "center",
      }
    }>
    Don\'t worry, only you can see your child\'s picture. If you feel uncomfortable using their photo, you can use any image you like.
  </Text>
  <View
    style={
      Object {
        "height": 36,
      }
    } />
  <TextInput
    autoCapitalize="words"
    autoCorrect={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="eg. Jen Smith"
    placeholderTextColor="rgba(75, 73, 73, 0.8)"
    style={
      Object {
        "backgroundColor": "#EEE",
        "borderRadius": 6,
        "color": "rgba(75, 73, 73, 1)",
        "fontSize": 16,
        "fontWeight": "normal",
        "height": 32,
        "marginBottom": 13,
        "marginHorizontal": 0,
        "padding": 6,
        "width": 718,
      }
    }
    underlineColorAndroid="rgba(255, 255, 255, 1)" />
  <View
    style={
      Object {
        "height": 16,
      }
    } />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Image
      source={1} />
  </View>
</View>
`;
